js/btcex.js

```diff
@@ -1275,7 +1305,61 @@ module.exports = class btcex extends Exchange {
+            } else {
+                const requestType = (reduceOnly) ? 'LONG' : 'SHORT';
+                request['position_side'] = requestType;
+            }
+            const stopPrice = this.safeNumber2 (params, 'triggerPrice', 'stopPrice');
+            let stopLossPrice = this.safeNumber (params, 'stopLossPrice');
+            let takeProfitPrice = this.safeNumber (params, 'takeProfitPrice');
+            const isStopLoss = this.safeValue (params, 'stopLoss');
+            const isTakeProfit = this.safeValue (params, 'takeProfit');
+            if (stopPrice) {
+                request['condition_type'] = 'STOP';
+                request['trigger_price'] = this.priceToPrecision (symbol, stopPrice);
+                request['trigger_price_type'] = 1;
+            } else if (stopLossPrice || takeProfitPrice) {
+                request['condition_type'] = 'STOP';
+                if (stopLossPrice) {
+                    request['trigger_price'] = this.priceToPrecision (symbol, stopLossPrice);
+                } else {
+                    request['trigger_price'] = this.priceToPrecision (symbol, takeProfitPrice);
+                }
+                request['reduce_only'] = true;
+                request['trigger_price_type'] = 1;
+            } else if (isStopLoss || isTakeProfit) {
+                if (isStopLoss) {
+                    stopLossPrice = this.safeNumber (isStopLoss, 'price');
+                    request['stop_loss_price'] = this.priceToPrecision (symbol, stopLossPrice);
+                    request['stop_loss_type'] = 1;
+                } else {
+                    takeProfitPrice = this.safeNumber (isTakeProfit, 'price');
+                    request['take_profit_price'] = this.priceToPrecision (symbol, takeProfitPrice);
+                    request['take_profit_type'] = 1;
+                }
+            }
+            if (type === 'market') {
+                if (side === 'buy') {
+                    const createMarketBuyOrderRequiresPrice = this.safeValue (this.options, 'createMarketBuyOrderRequiresPrice', false);
+                    if (createMarketBuyOrderRequiresPrice) {
+                        if (price === undefined) {
+                            throw new InvalidOrder (this.id + ' createOrder() requires a price argument for market buy orders on swap markets to calculate the total amount to spend (amount * price), alternatively set the createMarketBuyOrderRequiresPrice option to false and pass in the cost to spend into the amount parameter');
+                        } else {
+                            const amountString = this.numberToString (amount);
+                            const priceString = this.numberToString (price);
+                            const cost = this.parseNumber (Precise.stringMul (amountString, priceString));
+                            request['amount'] = this.priceToPrecision (symbol, cost);
+                            request['market_amount_order'] = true;
```

Comment:
what does this parameter do? It's not clear and even with `market_amount_order`, I didn't notice any difference

Reply:
```
Advanced order amount type, (Only for perpetuals), default: false. If set to true then the amount field means USDT value.
```
it changes the amount parameter to price in USDT for swaps only if createMarketBuyOrderRequiresPrice = True

Reply:
@Dan-krm in that case we should not throw if the user provides the amount in the base currency because we just need to flip that flag. If we should use that parameter if the user provides the cost inside params

Reply:
:heavy_check_mark: